src/renderers/dom/fiber/__tests__/ReactDOMFiber-test.js

```diff
@@ -1004,6 +1004,77 @@ describe('ReactDOMFiber', () => {
         container,
       );
     });
+
+    it('should not warn when rendering into an empty container', () => {
+      spyOn(console, 'error');
+      ReactDOM.render(<div>'foo'</div>, container);
+      expect(container.innerHTML).toBe("<div>'foo'</div>");
+      ReactDOM.render(null, container);
+      expect(container.innerHTML).toBe('');
+      expectDev(console.error.calls.count()).toBe(0);
+      ReactDOM.render(<div>'bar'</div>, container);
+      expect(container.innerHTML).toBe("<div>'bar'</div>");
+      expectDev(console.error.calls.count()).toBe(0);
+    });
+
+    it('should warn when replacing a container which was manually updated outside of React', () => {
+       spyOn(console, 'error');
+      // when not messing with the DOM outside of React
+      ReactDOM.render(<div key='1'>'foo'</div>, container);
+      ReactDOM.render(<div key='1'>'bar'</div>, container);
+      expect(container.innerHTML).toBe("<div>'bar'</div>");
+      // then we mess with the DOM before an update
+      // we know this will error - that is expected right now
+      expect(() => {
+        container.innerHTML = '<div>MEOW.</div>';
+        ReactDOM.render(<div key='2'>'baz'</div>, container);
+      }).toThrow();
```

Comment:
can you assert on the error text here to make sure the right error is thrown? I think passing a string to toThrow will work.